tests/friendly_errors/issue_7242.js

```diff
@@ -0,0 +1,9 @@
+// @flow
+
+type Ob = {|+foo: {| +bar: string |}|};
+declare var update: Ob;
+({ foo: { bar: 'valid' }, ...update }: Ob); // Ok
+
+type NestedOb = {|+foo: {| +bar: {| +foo: number |} |}|};
+declare var update2: NestedOb;
+({ foo: { bar: { foo: 123 } }, ...update2 }: NestedOb); // Ok
```

Comment:
not sure if firendly_errors folder is correct folder for these, but https://github.com/facebook/flow/blob/master/tests/friendly_errors/prop-variance.js is perhaps the closest to above tests that I found.